### PR TITLE
Better atomics annotations && a few false positive fixes

### PR DESCRIPTION
**Pull request by xairy**

### CODE REVIEW

File: arch/x86/include/asm/bitops.h [hunk 1]
```diff
@@ -166,7 +166,11 @@ static inline void __clear_bit(long nr, volatile unsigned long *addr)
  */
 static inline void __clear_bit_unlock(long nr, volatile unsigned long *addr)
 {
+#ifndef CONFIG_KTSAN
 	barrier();
+#else /* CONFIG_KTSAN */
+	ktsan_thread_fence(ktsan_memory_order_release);
```
Comment (dvyukov) [inline arch/x86/include/asm/bitops.h]: replace it with:

static inline void __clear_bit_unlock(long nr, volatile unsigned long *addr) {
	ktsan_atomic_store(addr, *(unsigned long*)addr & ^(1<<nr), ktsan_memory_order_release);
}

File: kernel/pid.c [hunk 2]
```diff
@@ -244,6 +244,9 @@ void put_pid(struct pid *pid)
 	ns = pid->numbers[pid->level].ns;
 	if ((atomic_read(&pid->count) == 1) ||
 	     atomic_dec_and_test(&pid->count)) {
+		/* Turn atomic_read into load acquire since ktsan
+		   doesn't understand control dependency. */
+		rmb();
```
Comment (dvyukov) [inline kernel/pid.c]: add #ifdef CONFIG_KTSAN

File: mm/page-writeback.c [hunk 3]
```diff
@@ -2785,7 +2785,14 @@ EXPORT_SYMBOL(__test_set_page_writeback);
  */
 int mapping_tagged(struct address_space *mapping, int tag)
 {
-	return radix_tree_tagged(&mapping->page_tree, tag);
+	int rv;
+	unsigned long flags;
+
+	spin_lock_irqsave(&mapping->tree_lock, flags);
```
Comment (dvyukov) [inline mm/page-writeback.c]: please add a comment explaining why the mutex was added

File: arch/x86/include/asm/bitops.h (hunk 1 above)
Comment (xairy) [inline arch/x86/include/asm/bitops.h]: Done.

File: kernel/pid.c (hunk 2 above)
Comment (xairy) [inline kernel/pid.c]: Done.

File: mm/page-writeback.c (hunk 3 above)
Comment (xairy) [inline mm/page-writeback.c]: Done.

**Comment by dvyukov** [comment]
LGTM

**Pull request merged**